src/parquet/util/bit-stream-utils.inline.h

```diff
@@ -86,34 +89,106 @@ inline bool BitWriter::PutVlqInt(uint32_t v) {
 }
 
 template <typename T>
+inline void GetValue_(
+  int num_bits, T* v, const uint8_t* buffer, int& bit_offset,
+  int& byte_offset, uint64_t& buffered_values, int max_bytes) {
```

Comment:
The Google C++ guide counterindicates mutable reference arguments: https://google.github.io/styleguide/cppguide.html#Reference_Arguments. Can you pass the mutable arguments as pointers (`uint64_t* buffered_values` would be more clear, anyway). You'll have to set the out values of `bit_offset` and `byte_offset` explicitly, but that is better than implicit mutation